bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -443,22 +443,19 @@ void ReplicaImp::onMessage<ClientRequestMsg>(std::unique_ptr<ClientRequestMsg> m
     std::ostringstream oss("ClientRequestMsg is invalid. ");
     oss << KVLOG(invalidClient, sentFromReplicaToNonPrimary);
     onReportAboutInvalidMessage(msg, oss.str().c_str());
-    delete msg;
+
     return;
   }
 
   if (readOnly) {
     if (activeExecutions_ > 0) {
       if (deferredRORequests_.size() < maxQueueSize_) {
         // We should handle span and deleting the message when we handle the deferred message
-        deferredRORequests_.push_back(msg);
+        deferredRORequests_.push_back(m.release());
```

Comment:
Why not
`deferredRORequests_.push_back(std::move(m));`
?

I've checked and it does make sense to extne the task into convertion of this container to hold unique_ptr. There is only one place for you to modift code, in ReplicaImp::handleDeferredRequests, so that we can reove the delete there as well.

Reply:
Done